### DISCUSSION TITLE
requestAnimationFrame vs SetTimeout

### DISCUSSION THREAD

**Discussion by franfrutos** [jspsych/jsPsych#2705] [General]
Hi! I'm new to programming online experiments using jsPsych and have been looking into time presentation issues. I've read that requestAnimationFrame is preferable to setTimeout because it allows the browser to synchronize with the frame painting process. 

Also, it looks like jsPsych uses setTimeout to time the presentation. I was wondering if anyone has managed to implement the requestAnimationFrame method to improve the time presentation.

Many thanks in advance!

**Comment by kurokida** [comment]
Hi @franfrutos ,
You might be interested in my plugin.
See https://jspsychophysics.hes.kyushu-u.ac.jp/

Best,
Daiichiro
